### PR TITLE
Work around protobuf issues by importing onnx first

### PR DESCRIPTION
**Pull request by anderspapitto**
So, I'm skeptical that this is actually the best fix, but let's discuss it

### CODE REVIEW

File: caffe2/python/onnx/backend.py [hunk 1]
```diff
@@ -15,6 +15,13 @@
 from subprocess import Popen, PIPE
 import zipfile
 
+# When onnx is built against a version of protobuf that is older than
+# that which is vendored with caffe2, onnx will crash if caffe2's
+# vendored protobuf is loaded first. We can work around this by
+# importing onnx first, which will cause it to go out and pick up the
+# system protobuf.
+import onnx.backend
```
[the remark this answers is not in the document]
Comment (houseroad) [reply caffe2/python/onnx/backend.py]: Yes, we can reproduce this. Try
`import caffe2.python.onnx.backend`.

[the remark this answers is not in the document]
Comment (yinghai) [reply caffe2/python/onnx/backend.py]: What does the crash log look like?

[the remark this answers is not in the document]
Comment (anderspapitto) [reply caffe2/python/onnx/backend.py]:
```
$ python -c 'import caffe2.python.onnx.backend'
[libprotobuf FATAL /home/andersj/onnx-dev/pytorch/third_party/protobuf/src/google/protobuf/stubs/common.cc:79] This program was compiled against version 2.6.1 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.5.0).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "google/protobuf/descriptor.pb.cc".)
terminate called after throwing an instance of 'google::protobuf::FatalException'
  what():  This program was compiled against version 2.6.1 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.5.0).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "google/protobuf/descriptor.pb.cc".)
Aborted (core dumped)
```

[the remark this answers is not in the document]
Comment (anderspapitto) [reply caffe2/python/onnx/backend.py]: yeah, the reproduction is to simply run that import command when the system version of protobuf is 2.6, and when caffe2 has BUILD_CUSTOM_PROTOBUF=ON (the default), and when the version of the python protobuf package is 3.5

If you just switch the version of the python protobuf package to 2.6, there's a different error:

```
$ python -c 'import caffe2.python.onnx.backend'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "caffe2/python/onnx/backend.py", line 19, in <module>
    from caffe2.python import core, workspace, rnn_cell, gru_cell
  File "caffe2/python/core.py", line 14, in <module>
    from caffe2.proto import caffe2_pb2
  File "caffe2/proto/caffe2_pb2.py", line 23, in <module>
    x18\x06 \x01(\x08\x12\x18\n\x10should_stop_blob\x18\t \x01(\t\x12\x11\n\tonly_once\x18\n \x01(\x08\x12\x18\n\x10\x63reate_workspace\x18\x0c \x01(\x08\x12 \n\x18num_concurrent_instances\x18\r \x01(\x05\"g\n\x07PlanDef\x12\x0c\n\x04name\x18\x01 \x01(\t\x12\x1f\n\x07network\x18\x02 \x03(\x0b\x32\x0e.caffe2.NetDef\x12-\n\x0e\x65xecution_step\x18\x03 \x03(\x0b\x32\x15.caffe2.ExecutionStep\"\xba\x01\n\tBlobProto\x12\x0c\n\x04name\x18\x01 \x01(\t\x12\x0c\n\x04type\x18\x02 \x01(\t\x12#\n\x06tensor\x18\x03 \x01(\x0b\x32\x13.caffe2.TensorProto\x12\x0f\n\x07\x63ontent\x18\x04 \x01(\x0c\x12%\n\x07qtensor\x18\x05 \x01(\x0b\x32\x14.caffe2.QTensorProto\x12\x1a\n\x12\x63ontent_num_chunks\x18\x06 \x01(\x05\x12\x18\n\x10\x63ontent_chunk_id\x18\x07 \x01(\x05\"K\n\rDBReaderProto\x12\x0c\n\x04name\x18\x01 \x01(\t\x12\x0e\n\x06source\x18\x02 \x01(\t\x12\x0f\n\x07\x64\x62_type\x18\x03 \x01(\t\x12\x0b\n\x03key\x18\x04 \x01(\t*|\n\nDeviceType\x12\x07\n\x03\x43PU\x10\x00\x12\x08\n\x04\x43UDA\x10\x01\x12\n\n\x06MKLDNN\x10\x02\x12\n\n\x06OPENGL\x10\x03\x12\n\n\x06OPENCL\x10\x04\x12!\n\x1d\x43OMPILE_TIME_MAX_DEVICE_TYPES\x10\x05\x12\x14\n\rONLY_FOR_TEST\x10\xc5\xde\xfb\t')
```

[the remark this answers is not in the document]
Comment (yinghai) [reply caffe2/python/onnx/backend.py]: I see. I think the current fix is the best we can do then...

**Review by yinghai — APPROVED** [review]

**Pull request merged**